org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/ConvertLambdaBlockToExpressionProposal.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.corrections.proposals;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Block;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.ExpressionStatement;
+import org.eclipse.jdt.core.dom.ReturnStatement;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+
+public class ConvertLambdaBlockToExpressionProposal extends LinkedCorrectionProposal {
+
+	private final Block lambdaBody;
+
+	public ConvertLambdaBlockToExpressionProposal(String name, String kind, Block node, ICompilationUnit cu, int relevance) {
+		super(name, kind, cu, null, relevance);
+		lambdaBody = node;
+	}
+
+	@Override
+	protected ASTRewrite getRewrite() throws CoreException {
+		ASTRewrite rewrite = ASTRewrite.create(lambdaBody.getAST());
```

Comment:
Not a huge fan of diverging from https://github.com/eclipse-jdt/eclipse.jdt.ui/blob/f8ac84e2d3303aa7f9d939e6d6023fc209b61661/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/fix/LambdaExpressionAndMethodRefCleanUp.java#L425-L435 but I guess this is fundamentally just replacing the body with the first expression, which won't be a block. Can you confirm comments remain in place ?